packages/ses/types.d.ts

```diff
@@ -33,6 +33,12 @@ export interface RepairOptions {
   overrideTaming?: 'moderate' | 'min' | 'severe';
   overrideDebug?: Array<string>;
   domainTaming?: 'safe' | 'unsafe';
+  /**
+   * none (default): do nothing.
+   *
+   * safe: make %IteratorPrototype%[@@iterator] to a funky accessor which ignores all assignments.
+   */
```

Comment:
I’m impressed that this is all that is necessary to compensate for the regenerator runtime. This looks a great deal like one of our “enablements” and maybe we can harness that.

I would like a test that embeds regenerator runtime as a shim to verify that this is sufficient.

Reply:
I'm not clear about what is "enablements", would you like to explain?

> I would like a test that embeds regenerator runtime as a shim to verify that this is sufficient.

Yes I have given the code.

```js
import './lockdown.umd.js'
lockdown({ regeneratorRuntimeTaming: 'safe', errorTaming: 'unsafe', consoleTaming: 'unsafe' })
const script = document.createElement('script')
script.src = 'https://cdn.jsdelivr.net/npm/regenerator-runtime@0.13.7/runtime.js'
document.head.appendChild(script)
```

Reply:
> I'm not clear about what is "enablements", would you like to explain?

"Fixes" enumerated cases of the override mistake by turning those data properties into accessor properties that emulate how a data property would have acted without the override mistake.

Please see 
https://github.com/endojs/endo/blob/master/packages/ses/src/enablements.js
https://github.com/endojs/endo/blob/master/packages/ses/src/enable-property-overrides.js
and, for example
https://github.com/endojs/endo/blob/master/packages/ses/src/enable-property-overrides.js

At first I liked @kriskowal suggestion, which would amount to adding the line
```js
    [iteratorSymbol]: true,
```
after line 175 https://github.com/endojs/endo/blob/59b756eb896c72a75f4c0773f964903364bf7e71/packages/ses/src/enablements.js#L175

However, then I realized that the problem that this PR fixes is not an assignment to a mutable object that inherits the property from a frozen object. Rather, it is a pointless assignment to the frozen prototype object itself, which cannot succeed. The enablement mechanism would still make this fail by throwing, as it must in the general case. This PR makes *that* assignment fail silently rather than throwing. We cannot make this the default behavior, because making an assignment silently fail is ***not*** safe. It causes control flow to proceed on paths that assume the assignment succeeded. When we introduced strict mode in ES5, that's why we made failed assignments throw. Because the ES3 sloppy behavior was ***unsafe***.

> I also want to consider automatically detecting the regenerator runtime and applying this mitigation without a feature flag.

In @Jack-Works 's example, the regenerator runtime is not initialized until ***after*** `lockdown`. If we take that as a constraint on the problem we're solving, I don't see a way to get the effect by automatically detecting, at least without violating a harder taboo: Making behavior of the primordials vary depending on what code gets loaded after `lockdown`. We clearly cannot do that.

So I am ok with the overall shape of this PR.

Reply:
I appreciate that the code snippet you gave above does serve as a manual test. But we should also have an automated ava test as part of the PR, so it becomes a perpetual part of CI. IMO, that test does not need to actually introduce a devDependency on a legacy version of regenerator runtime, though @kriskowal may disagree. (@kriskowal ?) The test I have in mind could just test the particular assignments in question, with a comment about which one is motivated by legacy regenerator runtime.